keps/sig-node/5055-device-taints-and-tolerations/README.md

```diff
@@ -0,0 +1,978 @@
+the devices that they manage in ResourceSlices. This information is used by the
+scheduler when selecting devices for user requests in ResourceClaims.
+
+With this KEP, DRA drivers can mark devices as tainted such that they won't be
+used for scheduling new pods. In addition, pods already running with access to
+a tainted device can be stopped automatically. Cluster administrators can do
+the same by creating a
+[ResourceSliceOverride](../5027-dra-admin-controlled-device-attributes] with a
+taint.
+
+Users can decide to ignore specific taints by adding tolerations to their
+ResourceClaim.
+
+## Motivation
+
+### Goals
+
+- Enable taking devices offline for maintenance while still allowing tests pods
+  to request and use those devices. Being able to do this one device at a time
+  minimizes service level disruption.
+
+- Enable users to decide whether they want to keep running a workload in a degraded
+  mode while a device is unhealthy or prefer to get pods rescheduled.
+
+### Non-Goals
+
+- Not part of the plan for alpha: developing a kubectl command for tainting devices.
+  This may be reconsidered.
+
+## Proposal
+
+### User Stories
+
+#### Degraded Devices
+
+A driver itself can detect problems which may or not be tolerable for
+workloads, like degraded performance due to overheating. Removing such devices
+from the ResourceSlice would unconditionally prevent using them for new
+pods. Instead, publishing with a taint informs users about this degradation and
+leaves them the choice whether the device is still usable enough to run pods.
+It also automates stopping pods which don't tolerate such a degradation.
+
+#### External Health Monitoring
+
+As cluster admin, I am deploying a vendor-provided DRA driver together with a
+separate monitoring component for hardware aspects that are not available or
+not supported by that DRA driver. When that component detects problems, it can
+check its policy configuration and decide to take devices offline by creating
+a ResourceSliceOverride with a taint for affected devices.
+
+### Risks and Mitigations
+
+A device can be identified by its names (`<driver name>/<pool name>/<device
+name>`) and/or by its attributes (for example, a unique ID). It was a conscious
+decision for core DRA to not require that the name is tied to one particular
+hardware instance to support hot-swapping. Admins might favor using the names
+whereas health monitoring might prefer to be specific and use a vendor-defined
+unique ID. Both is supported, which creates additional complexity.
+
+Without a kubectl extension similar to `kubectl taint nodes` the user
+experience for admins will be a bit challenging. They need to decide how to
+identify the device (by name or with a CEL expression), manually create a
+ResourceSliceOverride with a unique name, then remember to remove that
+ResourceSliceOverride again. For beta, support in `kubectl` for common
+operations may be needed.
```

Comment:
These two paragraphs are also new and discuss the usability aspect. While spelling it out, I noticed that selecting a device by name was in fact not specified yet, so I added it to `ResourceSliceOverride` - see class/driver/pool/device fields there.